app/scripts/controllers/reviewRegistration.js

```diff
@@ -280,4 +280,143 @@ angular.module('confRegistrationWebApp')
     $scope.hasPendingCheckPayment = function(payments){
       return _.some(payments, { paymentType: 'CHECK', status: 'PENDING' });
     };
+
+    ////// START EVENT-433 //////
+
+    // Get spouse registration
+    // End point: getSpouseRegistration = /conferences/conferenceId/registrations/spouse
+    ///// TEST DATA change end point url below to: /conferences/conferenceId/registrations/current /////
+    $http.get('conferences/' + conference.id + '/registrations/spouse')
+      .then(function(response) {
+        if (response.status === 200) {
+          $scope.spouseRegistration = response.data;
+
+          ///// TEST DATA /////
+          // $scope.spouseRegistration.completed = true;
+          // $scope.spouseRegistration.registrants[0].email = 'janeworner3@hotmail.com';
+          // $scope.spouseRegistration.registrants[0].firstName = 'Jane';
+          // $scope.spouseRegistration.registrants[0].lastName = 'Worner3';
+          ///// TEST DATA ////
+
+        }
+        else {
+          $scope.spouseRegistration = {};
+          console.log('No current and/or spouse registration found for user with ID.');
+        }
+      }).catch(function(response) {
+      $scope.spouseRegistration = {};
+      console.log('No current and/or spouse registration found for user with ID. Status = ' + response.status);
+      console.log(response.data);
+    })
+      .finally(function() {
+        checkSpouseRegistration();
+      });
+
+    //Parameters needed to retrieve all current registrants for conference.id
+    var queryParameters = {
+      block: [],
+      //page: 1,
+      //limit: 20,
+      orderBy: 'last_name',
+      order: 'ASC',
+      filter: '',
+      filterPayment: '',
+      filterRegType: '',
+      includeCheckedin: 'yes',
+      includeWithdrawn: 'yes',
+      includeIncomplete: 'yes'
+    };
+
+    //Get all current registrants based on conference.id and queryParameters
+    //then check if spouse is registered and display 'Spouse Registration' section if needed.
+    var checkSpouseRegistration = function() { // jshint ignore:line
+      $http.get('conferences/' + conference.id + '/registrations', {params: queryParameters})
+        .then(function (response) {
+          var _spouseSectionVisible = sessionStorage.getItem('spouseSectionVisible');
+          //var spouseIncludedOnCurrentRegistration = _.contains(_.map($scope.spouseRegistration.registrants, 'email'), $scope.currentRegistration.registrants[0].email);
+          var spouseIncludedOnCurrentRegistration = !_.isEmpty(_.intersection(_.map($scope.spouseRegistration.registrants, 'email'),
+            _.map(_.map(_.map(response.data.registrations, 'registrants'), 0), 'email')));
+
+          // 2. If the registrant does have a spouse who has already registered, then
+          if ($scope.spouseRegistration.completed) {
```

Comment:
I think I got all the changes you suggested made.
Ignore the /// TEST DATA /// section. 
I was trying to play around with the test data to actually test the logic.